inspirehep/modules/authors/views.py

```diff
@@ -225,26 +227,35 @@ def new():
     return render_template('authors/forms/new_form.html', form=form, **ctx)
 
 
+def _author_record_from_db(recid):
+    """Get author record from DB or return None if it is not there."""
+    try:
+        return get_db_record('aut', recid)
+    except RecordGetterError:
+        return None
+
+
+def _author_record_from_legacy(recid):
+    """Get author record from Legacy or return None if it is not there."""
+    try:
+        url = get_legacy_url_for_recid(recid) + '/export/xm'
+        xml_response = requests.get(url)
+        xml = fromstring(xml_response.content)
+        xml_record = xml.xpath("//*[local-name() = 'record']")[0]
+        return marcxml2record(tostring(xml_record))
+    except requests.exceptions.RequestException:
```

Comment:
the `try` `except` conflates multiple things, but only catches `RequestException`. What happens for non-existent `recid` or a wrong `recid`, e.g. a non-integer? 
e.g.
```
$ curl http://inspirehep.net/record/9999999/export/xm
<?xml version="1.0" encoding="UTF-8"?>
<collection xmlns="http://www.loc.gov/MARC21/slim">

</collection>
```
then you get an `IndexError` on line 244

and for `recid = 9999O` you get a status `404` and HTML response which results in `XMLSyntaxError`.

I recommend to check xml_response.status_code first.